extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -59,27 +62,40 @@ export abstract class Validation {
 		return this._valueGetter();
 	}
 
-	protected getVariableValue(variable: string): Promise<ValidationValueType> {
-		return this._variableValueGetter!(variable);
+	protected getTargetValue(variable: string): Promise<ValidationValueType> {
+		return this._targetValueGetter!(variable);
 	}
 
-	constructor(validation: ValidationInfo, protected _valueGetter: ValueGetter, protected _variableValueGetter?: VariableValueGetter) {
+	protected async onValidation(isValid: boolean): Promise<void> {
+		await this._onValidation(isValid);
+	}
+
+	constructor(validation: ValidationInfo, protected _onValidation: onValidation, protected _valueGetter: ValueGetter, protected _targetValueGetter?: TargetValueGetter, protected _onTargetValidityChangedGetter?: OnTargetValidityChangedGetter, protected _onNewDisposableCreated?: (disposable: vscode.Disposable) => void) {
 		this._description = validation.description;
 	}
+
+	protected isUndefinedOrEmpty(value: ValidationValueType): boolean {
+		return value === undefined || (typeof value === 'string' && value.length === 0);
+	}
 }
 
 export class IntegerValidation extends Validation {
-	constructor(validation: IntegerValidationInfo, valueGetter: ValueGetter) {
-		super(validation, valueGetter);
+	constructor(validation: IntegerValidationInfo, validationMessageUpdater: onValidation, valueGetter: ValueGetter) {
+		super(validation, validationMessageUpdater, valueGetter);
 	}
 
 	private async isInteger(): Promise<boolean> {
 		const value = await this.getValue();
-		return (typeof value === 'string') ? Number.isInteger(parseFloat(value)) : Number.isInteger(value);
+		return (this.isUndefinedOrEmpty(value))
```

Comment:
This should be made much more clear that it returns true for undefined or empty values as well. Probably just a doc comment is good enough but a function name update wouldn't be the worst either

Reply:
Ping

Reply:
What returns undefined/empty value? isInteger returns only a promise to a boolean. If you see it returning something else then I am missing a bug and please let me know specifically where isInteger returns undefined or empty value.

Reply:
changed function name to isIntegerOrUndefinedOrEmpty